### PR TITLE
Adjust coordinate system for drag and drop with vertical tab strip

### PR DESCRIPTION
**Pull request by sangwoo108**
* Make sure that moving the window with tabs doesn't jitter - window shouldn't move to other point abrubtly.
* Make sure that detached tab is right below the mouse cursor.
* Expand vertical tab strip temporarily during drag-and-drop session.

Resolves https://github.com/brave/brave-browser/issues/26770

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### Manual
* When dragging tabs into another tabstrip, the tabstrip should be exapnded
* When dragging all tabs in a window, window should move smoothly without jumping to other place
* When detaching tabs, the tabs should be under cursor.

### CODE REVIEW

File: chromium_src/chrome/browser/ui/views/tabs/tab_drag_controller.cc [hunk 1]
```diff
@@ -52,6 +63,7 @@ void TabDragController::Init(TabDragContext* source_context,
   source_view_offset = mouse_offset.y();
   start_point_in_screen_ = gfx::Point(mouse_offset.x(), source_view_offset);
   views::View::ConvertPointToScreen(source_view, &start_point_in_screen_);
+  LOG(ERROR) << start_point_in_screen_.ToString();
```
Comment (iefremov) [inline chromium_src/chrome/browser/ui/views/tabs/tab_drag_controller.cc:66]: cleanup

**Review by iefremov — APPROVED** [review]
chromium_src lgtm

File: chromium_src/chrome/browser/ui/views/tabs/tab_drag_controller.cc (hunk 1 above)
Comment (sangwoo108) [reply chromium_src/chrome/browser/ui/views/tabs/tab_drag_controller.cc:66]: oops😅, thanks

File: browser/ui/views/frame/vertical_tab_strip_region_view.cc [hunk 2]
```diff
@@ -315,6 +315,28 @@ void VerticalTabStripRegionView::SetState(State state) {
   PreferredSizeChanged();
 }
 
+VerticalTabStripRegionView::ScopedStateResetter
+VerticalTabStripRegionView::ExpandTabStripForDragging() {
+  if (state_ == State::kExpanded)
+    return {};
+
+  auto resetter = std::make_unique<base::ScopedClosureRunner>(
```
Comment (simonhong) [inline browser/ui/views/frame/vertical_tab_strip_region_view.cc:323]: Oh, this resetter looks cool!

**Review by simonhong — APPROVED** [review]
++ 👍🏼

File: browser/ui/views/frame/vertical_tab_strip_region_view.cc (hunk 2 above)
Comment (sangwoo108) [reply browser/ui/views/frame/vertical_tab_strip_region_view.cc:323]: Yeah, it's perfect for this kind of mementor pattern!

**Pull request merged**